airbyte-cdk/python/airbyte_cdk/sources/streams/concurrent/thread_based_concurrent_stream.py

```diff
@@ -25,7 +25,7 @@
 
 class ThreadBasedConcurrentStream(AbstractStream):
 
-    DEFAULT_TIMEOUT_SECONDS = 300
+    DEFAULT_TIMEOUT_SECONDS = 900
```

Comment:
One of the syncs I tried failed with a 5 minutes timeout. It succeeded with 10 minutes, but increasing it to 15 so we have an additional buffer

Reply:
Does that mean that a partition took more than 5 minutes to be fetched? I'm not that surprised given that slices can be `[None]` but it make me wonder if this timeout shouldn't account for a whole sync time or if we need heartbeat of some sorts